board.es6

```diff
@@ -266,30 +224,74 @@ class Board {
     return Board.parseTeam( this.pieceObject(position) )=== Board.WHITE && Board.parseSpecies( this.pieceObject(position) ) === Board.PAWN
   }
 
-  blackPawnDoubleSteppedFrom(position){
-    return this.previousLayouts.length && this.positionEmpty(position) && Board.parseTeam( this.pieceObjectFromLastLayout(position) ) === Board.BLACK && Board.parseSpecies( this.pieceObjectFromLastLayout(position) ) === Board.PAWN
+  blackPawnDoubleSteppedTo(position){// only to be called if already know the black pawn is at rank 4 and in "position"
+    var result;
+    let blackMoves = this.movesNotationFor(Board.BLACK),
+      square = Board.gridCalculator(position),
+      hypotheticalSingleStepSquare = square[0] + '6'; //e.g. if the double step would've been to a4, then the single step it may have taken would've been to a3
+    if( blackMoves[blackMoves.length -1] === square ){
+      result = true;
+    } else {
+      return false //this means that even if it did at some point double step, it wasn't the last move to occur
+    }
+    for(let i = 0; i < blackMoves; i < blackMoves.length){
+      if( blackMoves[i] === hypotheticalSingleStepSquare || blackMoves[i] === hypotheticalSingleStepSquare + "+" ){
+        result = false;
+        break
+      }
+    }
+    return result
   }
 
-  whitePawnDoubleSteppedFrom(position){
-    return this.previousLayouts.length && this.positionEmpty(position) && Board.parseTeam( this.pieceObjectFromLastLayout(position) ) === Board.WHITE && Board.parseSpecies( this.pieceObjectFromLastLayout(position) ) === Board.PAWN
+
+  whitePawnDoubleSteppedTo(position){// only to be called if already know the white pawn is at rank 4 and in "position"
+    var result;
+    let whiteMoves = this.movesNotationFor(Board.WHITE),
+      square = Board.gridCalculator(position),
+      hypotheticalSingleStepSquare = square[0] + '3'; //e.g. if the double step would've been to a4, then the single step it may have taken would've been to a3
+    if( whiteMoves[whiteMoves.length -1] === square ){
+      result = true;
+    } else {
+      return false //this means that even if it did at some point double step, it wasn't the last move to occur
+    }
+    for(let i = 0; i < whiteMoves; i < whiteMoves.length){
+      if( whiteMoves[i] === hypotheticalSingleStepSquare || whiteMoves[i] === hypotheticalSingleStepSquare + "+" ){
+        result = false;
+        break
+      }
+    }
+    return result
+  }
+
+  movesNotationFor(team){
+    let teamMoves = [];
+    if( team === Board.WHITE ){
+      var initialElement = 0
+    } else if (team === Board.BLACK ){
+      var initialElement = 1
+    } else {
+      alert( "bad input for board.movesNotationFor: " + team )
```

Comment:
Try to avoid alerts. In this case, you should be throwing an error. Longer term, you should be using something like rollbar or Sentry to track JavaScript errors. Let me know if you'd like more information about that.